File: TodoList.WebClient/scripts/task/taskController.js
angular.module('todolist').controller('taskController', [
    '$scope', '$location', 'taskService', 'projectService', function ($scope, $location, taskService, projectService) {
        $scope.task = {Project:null};
        $scope.projects = [];


        $scope.save = function () {

            $scope.task.DueDate = $scope.task.DueDate.toDateString();
        $scope.task.ProjectId = $scope.task.Project.Id;
        console.log($scope.task);

        taskService.save($scope.task).then(function(response) {

            console.log(response);
            $scope.project = {};
            $location.path('/');
        }, function(error) {
            alert(error.statusText);
        });
    }

        var   loadProjects= function() {
            projectService.getAll().then(function(response) {
                if (response.IsSuccess) {
                    $scope.projects = response.Data;
                } else {
                    alert(response.Message);
                }
            }, function(error) {
                console.log(error);
            });

        }

        function init  () {

            loadProjects();

        }

        init();

    }
]);

File: TodoList.ApiApplication/Controllers/TaskController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Web.Http;
using TodoList.DataAccess;
using TodoList.Service;
using TodoList.ViewModel;

namespace TodoList.ApiApplication.Controllers
{
    public class TaskController : ApiController
    {
        private TaskService service = new TaskService();

        public ResponseModel Get(int projectId, string sortOn="", string sortBy="")
        {
            ResponseModel response;
            try
            {
                List<TaskViewModel> tasks = projectId == 0 ? service.GetAll() : service.GetAllByProject(projectId, sortOn, sortBy);

                response=new ResponseModel(tasks);
            }
            catch (Exception exception)
            {
                response=new ResponseModel(null,false,"Error Occurred",exception);
            }
            return response;
        }

        public ResponseModel GetDetail(int id)
        {
            ResponseModel response;
            try
            {
                if (id>0)
                {
                    Task task = service.GetById(id);

                    response = new ResponseModel(task);
                }

                else
                {
                    response=new ResponseModel(isSuccess:false,message:"Id can not be zero");
                }
                
            }
            catch (Exception exception)
            {
                response = new ResponseModel(null, false, "Error Occurred", exception);
            }
            return response;
        }

        public ResponseModel Post(Task task)
        {
            ResponseModel response;
            try
            {
                task.Project = null;
                int id = service.Save(task);
                response = id>0 ? new ResponseModel(id) : new ResponseModel(null,false,"Couldn't Save");
            }
            catch (Exception exception)
            {
                response=new ResponseModel(null,false,"Error Occurred",exception);
            }
            return response;
        }
        public ResponseModel Delete(int id)
        {
            ResponseModel response;
            try
            {
                bool deleted = service.Delete(id);
                response = deleted ? new ResponseModel(id) : new ResponseModel(null, false, "Couldn't delete");
            }
            catch (Exception exception)
            {
                response = new ResponseModel(null, false, "Error Ocured", exception);
            }
            return response;
        }
    }
}
